### ISSUE TITLE
ET loses connection to server and cannot reconnect

### ISSUE THREAD

**Issue by smgallo** [MisterTea/EternalTerminal#146]
I am experiencing an issue where ET will behave as expected much of the time throughout the day, but occasionally it is not able to successfully reconnect with the server after my laptop resumes from sleep or it simply loses the connection after sitting idle for some period of time. Other ET connections that use the same host sometimes may remain active while some fail. I have run ET with the --v=9 option and attached the logs.

et version 5.1.0

Client: MacBook Pro 2018, 10.14 (also happened in 10.13), Darwin bobbiedarin.xxx.xxx 18.0.0 Darwin Kernel Version 18.0.0: Wed Aug 22 20:13:40 PDT 2018; root:xnu-4903.201.2~1/RELEASE_X86_64 x86_64

Server: Ubuntu 16.04.05, Linux smgallo.xxx.xxx 4.4.0-130-generic #156-Ubuntu SMP Thu Jun 14 08:53:28 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

The event occurs at 2018-10-03 21:47:37 in the attached client logs, although the server logs seem to show that there is a last keepalive event at 2018-10-03 16:35:38 (which is when I left the office for the day) and a recovery event starting at 2018-10-03 17:01:09 but I did not open my laptop again until 2018-10-03 21:47.

[etclientLogs.tar.gz](https://github.com/MisterTea/EternalTerminal/files/2446401/etclientLogs.tar.gz)
[etserverLogs.tar.gz](https://github.com/MisterTea/EternalTerminal/files/2446402/etserverLogs.tar.gz)

The client connection is still running on the server:
```
smgallo@smgallo:/tmp$ ps -ef|grep etter
smgallo   8493     1  0 Oct03 ?        00:05:06 etterminal --idpasskeyfile=/tmp/et-server.3w8QLcmgYurJ --v=9
```

And on the client:
```
bobbiedarin:~$ ps -ef|grep et|grep smgallo.xxx
  501  7659  2165   0  9:52PM ttys000    0:11.45 et --v=9 smgallo.xxx.xxx
```

**Comment by MisterTea** [comment]
Hey!  Yep, this is high pri for me.  I created a set of tests for ET and ET isn't passing them yet :-).  Once ET passes them, the connection issues should be resolved.  You can follow here: https://github.com/MisterTea/EternalTerminal/pull/142

**Comment by smgallo** [comment]
Great, let me know if there's any testing that you'd like me to do.

**Comment by MisterTea** [comment]
Please try latest master (update both client and server) and see if the problem is fixed, thanks

**Comment by MisterTea** [comment]
Fixed in 5.1.6
